### ISSUE TITLE
[Sybil Report] - 74

### ISSUE THREAD

**Issue by uytre4** [LayerZero-Labs/sybil-report#1832]
---
name: Sybil Report
about: Report Sybil Activity on LayerZero
title: "[Sybil Report]"
labels: under review
assignees: LayerZero-GH

---

# Reported Addresses<br>

0x6cce78ea37236f276ce5e36d6ae072ec3a73f41c<br>0xfd5ac9fac26300d7fbf9ef83cf4f33b629d6e681<br>0xa748670596527d7f1b5975a7337c4b63075075d0<br>0x7653b93c53b154fd0323ad04dca2120a3ddac3bd<br>0x9d1ff88261193013ecdeb9986ff3090fa9fbe9e4<br>0x87f613b02bdc4fcd55ae23db27308afa2d3aa2df<br>0xc11514a5b2dbca0ff026e3134273b846493613c5<br>0x2b37c953dc280da171f1399b9f4bc59dee666cac<br>0xa0ddb2cc70a2ddc6de7a4559a9fa11f30f0c7c6f<br>0x3ceab42d2b460eaf308d178afb63eeb27340d762<br>0x007883a6b08c459b98a68b2b571fdb3ed0e72cd0<br>0x0e313d0fd05344aebbeb64288ef6961d1f93cafe<br>0xbb2f7d6116c31038bb8a8ca572e91ad27ef0da82<br>0x2f26b03eaa6c65467f69e4ac00583ebcac1b13c3<br>0x520167d02ce3175e9812b9ea2e8c75b10c2dcbdc<br>0x4d2903c6ea481b798ebc0a21402d5212282baa34<br>0x17f98a5c060107e7ab4d47eb73d1eadef957aa21<br>0x4e8b005c13ebbb1df49603fda0b1d1a14162d877<br>0x3d768d95a23d5bf3cb88363fffa10e7b88621243<br>0x104bf7d97bfe250af3f52599c55a3dcd9bf207a1<br>0xa313b5afea4975d92576c9448d6311909a45ee1a<br>0xa92a0f2565f0fc55ba170117532436f5e70c60a9<br>0x433457af04437d40d953660c5b33e8b74d80de84<br>0x0514ea7df1ca3778504b5426dc82640255229a5b<br>0x98ee42b804856e8ac010c19c257e2d0a5269046d<br>0xaa09bedfa8890f36bcbd678517a1379d4482aecc<br>0xefd330e3de3fa518d369ad67e5de97161c6091c7

# Description<br>

This cluster contains 27 addresses, each of which corresponds to all provided statistics and can be easily checked manually at https://layerzeroscan.com/.<br><br>
All addresses in the cluster use the same networks: [Arbitrum, Arbitrum Nova, Base, Celo Mainnet, Conflux eSpace, Fuse Mainnet, Kava, Klaytn Mainnet Cypress, Polygon, Zora, zkSync Era Mainnet].<br>Cluster-averaged correlation of vectors of used networks: 0.9953544800226508.<br><br>Cluster median correlation of the vectors of used networks: 0.9967008539262628.<br>
All addresses in the cluster use the same protocols: [Merkly].<br>
All addresses in the cluster have from 225434.025 to 314960.056 (avg 253429.711) seconds interval between transactions within the cluster.<br>Similarity rate across the cluster by interval between transactions: 0.908<br><br>
All addresses in the cluster have from 35 to 42 (avg 38.815) transactions count.<br>Similarity rate across the cluster by number of transactions: 0.948.<br><br>
All addresses in the cluster have from 35 to 42 (avg 38.741) unique* transactions count.<br>Similarity rate across the cluster by number of unique* transactions: 0.949.<br><br>
All addresses in the cluster have 0.0 share of transactions with sequentially* used networks.<br>Similarity rate across the cluster by share of transactions with sequentially* used networks: 1.<br><br>
All cluster addresses share a great deal of similarity in other ways that can be found at https://debank.com/profile/.<br>All addresses in the cluster have from 1695641888 to 1695682406 (avg 1695649160.519) born timestamp of the first transaction among all blockchains.<br>Similarity rate across the cluster by accounts born timestamp: 1.0.<br><br>

All of the above statistics confirm that the clustering was done correctly and all attached addresses are Sybils. Address transactions are performed on the same days, at approximately the same time, using the same protocols and networks. The transactions were probably performed by a script or sequentially manually on the accounts.<br><br>

# Detailed Methodology & Walkthrough<br>

1. data processing:<br>
1.1 Removing duplicates.<br>
1.2 Filling empty values.<br>
1.3 Generation of user and transaction datasets.<br>
2. Collecting and analyzing statistics:<br>
2.1. Obtaining mean and median values for number of transactions, number of unique* transactions, interval between transactions, networks and protocols used, transactions with sequentially* used networks.<br>
2.2 First attempts at clustering, selection of the best criteria and normalization of the data. Data reflecting one criterion and not paired with other criteria were not normalized, and other data were presented from 0 to 1.<br>
3. Multistage clustering. For clustering, model sklearn.cluster.DBSCAN was mainly used because it does not require specifying the number of clusters, can be executed on multiple CPU cores at once, and has flexible tuning.<br>
3.1 Clustering by protocols and networks, and then forming subclusters by transactions. For clustering by networks, a vector was constructed for each address to represent the number of times the user used each network. For further clustering by transactions, a vector was generated for each address reflecting the number of user transactions for every 4 hours (this proved to be the best time to optimize the computation and not degrade the clustering results). Clustering was performed with parameters `eps=2`, `min_samples=20`.<br>
3.2 Clustering by normalized statistics. Clustering on normalized data was performed with parameters `eps=0.3`, `min_samples=20` or `eps=0.6`.<br>
3.3 Clustering using additional metrics taken from debank. For example, account birth timestamp, all networks used by the account and also account balances in networks. The timestamp was used to select accounts created around the same time, for which further clustering by transaction similarity was performed. `eps=4` and `min_samples=20` was used.<br>
3.4 Searching for cluster intersections. The search for cluster intersections was performed using set() & set() to search for common addresses.<br>
3.5 Plotting. For each address, a graph was plotted to visualize the relationship with other addresses in the cluster.<br>
3.6 Manual address checks, checks via https://layerzeroscan.com/ and crosschecking crosschain statistics via https://debank.com/profile/.<br>

The code snippets used in the clustering are here: https://github.com/shsh-sh/hlibys.<br>
Most of the work was done by doing a long and deep analysis, brute force search of the best criteria, manual cluster checking and making sure that the clusters found contained only Sybil addresses.<br>

Normalizing the data, constructing correct vectors, partitioning into clusters and subclusters in two different ways and further searching for address overlaps in subclusters and checking against statistical data can ensure that the submitted addresses are Sybils driven.<br>

Explanation:<br>
* Unique transactions - transactions of one address with unique `SOURCE_TRANSACTION_HASH`. Used to ignore transactions that were made through one protocol to many networks simultaneously.<br>
* Transactions with sequentially used networks - successive transactions in which `SOURCE_CHAIN` is equal to `DESTINATION_CHAIN` of the previous transaction.<br>

# Reward Address (If Eligible)<br>
0x7DAB82514de70c45B459E884EBe57Ac3d91eeE09<br>
